engine/src/main/java/org/terasology/logic/common/RetainComponentsComponent.java

```diff
@@ -0,0 +1,11 @@
+package org.terasology.logic.common;
+
+import com.google.api.client.util.Sets;
+import org.terasology.entitySystem.Component;
+
+import java.util.Collections;
+import java.util.Set;
+
+public class RetainComponentsComponent implements Component {
+    public Set<Class<? extends Component>> components = Sets.newHashSet();
```

Comment:
We should make sure that this works with de-/serialization, e.g., check whether we can specify this also via a prefab...